From fa39cb0d4866a4893f6d346ca61bb3b1c0c18397 Mon Sep 17 00:00:00 2001
Message-ID: <fa39cb0d4866a4893f6d346ca61bb3b1c0c18397.1736059919.git.divya@subvertising.org>
In-Reply-To: <cover.1736059918.git.divya@subvertising.org>
References: <cover.1736059918.git.divya@subvertising.org>
From: Divya Ranjan <divya@subvertising.org>
Date: Sat, 4 Jan 2025 20:55:36 +0000
Subject: [PATCH 12/44] gnu: Add rust-wrapcenum-derive-0.4.

* gnu/packages/crates-io.scm (rust-wrapcenum-derive-0.4): New variable.
---
 gnu/packages/crates-io.scm | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/gnu/packages/crates-io.scm b/gnu/packages/crates-io.scm
index 1a5259c10c..a9c197c898 100644
--- a/gnu/packages/crates-io.scm
+++ b/gnu/packages/crates-io.scm
@@ -91937,6 +91937,29 @@ (define-public rust-wl-nl80211-0.2
      "This package provides Linux kernel wireless(802.11) netlink Library.")
     (license license:expat)))
 
+(define-public rust-wrapcenum-derive-0.4
+  (package
+    (name "rust-wrapcenum-derive")
+    (version "0.4.1")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (crate-uri "wrapcenum-derive" version))
+       (file-name (string-append name "-" version ".tar.gz"))
+       (sha256
+        (base32 "0vi67wxkpw40cch5jx010wvwf0rjc5f13yxfzd60acimadcz4vx7"))))
+    (build-system cargo-build-system)
+    (arguments
+     `(#:cargo-inputs (("rust-darling" ,rust-darling-0.20)
+                       ("rust-proc-macro2" ,rust-proc-macro2-1)
+                       ("rust-quote" ,rust-quote-1)
+                       ("rust-syn" ,rust-syn-2))))
+    (home-page "https://github.com/Cldfire/wrapcenum-derive")
+    (synopsis "Derive macro for nvml-wrapper, not for general use")
+    (description
+     "This package provides Derive macro for nvml-wrapper, not for general use.")
+    (license (list license:expat license:asl2.0))))
+
 (define-public rust-write16-1
   (package
     (name "rust-write16")
-- 
2.47.1

